spring-data-mongodb/src/test/java/org/springframework/data/mongodb/core/query/FieldUnitTests.java

```diff
@@ -85,4 +86,18 @@ void overriddenExclusionMethodsCreateEqualFields() {
 
 		assertThat(left).isEqualTo(right);
 	}
+
+	@Test
+	void assertDifferentHashCodesForExcludeAndIncludeQueries() {
+
+		Query queryWithExclude = new Query();
+		queryWithExclude.fields().exclude("key1");
+		queryWithExclude.fields().exclude("key2");
+
+		Query queryWithInclude = new Query();
+		queryWithInclude.fields().include("key1");
+		queryWithInclude.fields().include("key2");
+
+		assertThat(queryWithExclude.hashCode()).isNotEqualTo(queryWithInclude.hashCode());
```

Comment:
Before the fix, the test above failed.
<img width="1428" alt="image" src="https://github.com/user-attachments/assets/f053e0e1-c61c-40cc-8fa1-d783b7fe0f6c">